src/loaders/genome_collection/parse_biolog_data.py

```diff
@@ -0,0 +1,187 @@
+import uuid
+from pathlib import Path
+
+import pandas as pd
+
+from src.common.product_models.heatmap_common_models import (
+    ColumnType,
+    FIELD_HEATMAP_CELL_ID,
+    FIELD_HEATMAP_CELL_VALUE,
+    FIELD_HEATMAP_COL_ID,
+    FIELD_HEATMAP_COLUMNS,
+    FIELD_HEATMAP_DESCR,
+    FIELD_HEATMAP_MAX_VALUE,
+    FIELD_HEATMAP_MIN_VALUE,
+    FIELD_HEATMAP_NAME,
+    FIELD_HEATMAP_ROW_CELLS,
+    FIELD_HEATMAP_TYPE,
+    FIELD_HEATMAP_VALUES,
+    FIELD_HEATMAP_CATEGORY,
+    FIELD_HEATMAP_CATEGORIES,
+)
+from src.common.storage.collection_and_field_names import (
+    COLL_MICROTRAIT_CELLS,
+    COLL_MICROTRAIT_DATA,
+    COLL_MICROTRAIT_META,
+    FLD_ARANGO_KEY,
+    FLD_COLLECTION_ID,
+    FLD_LOAD_VERSION,
+)
+from src.common.storage.db_doc_conversions import collection_load_version_key
+from src.common.storage.field_names import FLD_KBASE_ID
+from src.loaders.common import loader_common_names
+from src.loaders.common.loader_helper import init_row_doc, create_import_files
+from src.loaders.genome_collection.parse_tool_results import HEATMAP_FILE_ROOT
+
+
+def _read_excel_as_df(excel_file: Path, sheet_name=0) -> pd.DataFrame:
+    # Read the Excel file into a DataFrame
+
+    try:
+        df = pd.read_excel(excel_file, sheet_name=sheet_name)
+
+        return df
+
+    except FileNotFoundError:
+        print(f"File '{excel_file}' not found.")
+
+
+def _read_biolog_data(biolog_data_file: Path) -> pd.DataFrame:
+    # Read the Biolog data file into a DataFrame and do some data cleaning
+
+    data_df = _read_excel_as_df(biolog_data_file)
+
+    # NOTE: below logic is specific to the Biolog data file (PMI strain BiologSummary.xlsx)
+    # link to the file: https://docs.google.com/spreadsheets/d/1QmC6UHWOEVfpmrveRBl_izictbNeN1PA/edit#gid=1135979967
+
+    # Drop unused columns
+    data_df = data_df.drop(columns=['# pos. strains'])
+
+    # Make the first row ('strain designation') the column names
+    data_df.columns = data_df.loc[1]
+
+    # Drop unused rows ('total cpds. Utilized', 'strain designation' and unused rows at the end.)
+    data_df = data_df.drop(index=[0, 1, 194, 195, 196, 197, 198])
+
+    # Make the first column ('strain designation') the index
+    data_df = data_df.set_index('strain designation')
+
+    # Transpose the DataFrame
+    data_df.index.name = None
+    data_df.columns.name = None
```

Comment:
Why are these lines necessary?

Reply:
removed. With or without, the result jsonl files are the same.

Reply:
:+1: